pkg/cmd/root/help_reference.go

```diff
@@ -0,0 +1,82 @@
+package root
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/markdown"
+	"github.com/spf13/cobra"
+)
+
+func referenceHelpFn(cmd *cobra.Command, io *iostreams.IOStreams) func() string {
+	cs := io.ColorScheme()
+	return func() string {
+		reftext := "# gh reference\n\n"
+
+		for _, c := range cmd.Commands() {
+			reftext += cmdRef(cs, c, "")
+		}
+
+		style := markdown.GetStyle(io.DetectTerminalTheme())
+		md, err := markdown.RenderWrap(reftext, style, io.TerminalWidth())
+		if err != nil {
+			return reftext
+		}
+
+		return md
+	}
+}
+
+func cmdRef(cs *iostreams.ColorScheme, cmd *cobra.Command, parent string) string {
+	ref := ""
+
+	if cmd.Hidden {
```

Comment:
As this is a reference for all of `gh` I am wondering if we should include the `hidden` commands also?

Reply:
I think my preference is no. If we choose to hide a command for some reason, that's intentional and therefore isn't intended to be part of the documentation. Definitely open to changing my mind on that, but part of the fun of having a few easter eggs is finding them.